lib/button-control/src/button.cpp

```diff
@@ -0,0 +1,58 @@
+#include "Button.hpp"
+
+Button::Button(int buttonNum)
+{
+  switch (buttonNum)
+  {
+  case 0:
+    pinNum = 3;
+    break;
+  case 1:
+    pinNum = 2;
+    break;
+  default:
+    // error
+    break;
+  }
+
+  pinMode(pinNum, OUTPUT);
+  press_time = 0;
+  release_time = 0;
+  button_active = false;
+
+}
+
+Button::~Button()
+{
+}
+
+Press Button::button_pressed()
+{
+  Press output = NOT_PRESSED;
+  int readState = digitalRead(pinNum);
+
+  if (readState == high_button_state && (button_active == false))
+  {
+    press_time = millis();
+    button_active = true;
+  }
+  else if ((readState != high_button_state) && (button_active == true))
+  {
+    release_time = millis();
+    button_active = false;
+  }
+  if (release_time > 0 && press_time > 0)
+  {
+
+    if (millis() - press_time >= debounce_delay)
+    {
+      unsigned long time_diff = release_time - press_time < 0 ? 0 : release_time - press_time; 
+      output = time_diff > long_press ? PRESSED_LONG :
+              ((time_diff > medium_press) && (time_diff <= long_press)) ? PRESSED_MEDIUM :
+              ((time_diff > 0) && (time_diff <= medium_press)) ? PRESSED_SHORT : NOT_PRESSED;
```

Comment:
You can split this to `if` but i don't see too much problem with this if its stay in that form. 
But what i would change for sure you can cut this `if` because if  `time_diff > long_press` is false it's for sure `time_diff <= long_press` so you don't need to write it in else statement. Same to `medium_press`.

Reply:
Yes, it looks really complicated and too long.

Reply:
You can create a separate private methods for some `if` statements, for example: `check_long()`, `check_short()`, `check_medium()` etc.